utils/cloud_cleanup/azure/clean_azure.py

```diff
@@ -0,0 +1,187 @@
+        except Exception as exc:  # pylint: disable=broad-except
+            LOGGER.info("Failed to update RG tags: {resource_group.name} with exception: {exc}".format(
+                resource_group=resource_group, exc=exc))
+    return creation_time
+
+
+def get_keep_action(v_m) -> Callable:
+    return v_m.tags.get('keep_action', "stop").lower() if v_m.tags else "stop"
+
+
+def should_keep(creation_time, keep_hours):
+    if keep_hours <= 0:
+        return True
+    try:
+        keep_date = creation_time + timedelta(hours=keep_hours)
+        return datetime.utcnow() < keep_date
+    except (TypeError, ValueError):
+        return True
+
+
+def delete_virtual_machine(resource_group_name, vm_name, dry_run=False):
+    if dry_run:
+        LOGGER.info("[DRY RUN] Would delete VM: {vm_name} in resource group: {resource_group_name}".format(
+            vm_name=vm_name, resource_group_name=resource_group_name))
+    else:
+        LOGGER.info("Deleting VM: {vm_name} in resource group: {resource_group_name}".format(
+            vm_name=vm_name, resource_group_name=resource_group_name))
+        try:
+            compute_client.virtual_machines.begin_delete(resource_group_name, vm_name)
+        except Exception as exc:  # pylint: disable=broad-except
+            LOGGER.info(
+                "Failed to delete VM: {vm_name} in resource group: {resource_group_name} with exception: {exc}"
+                .format(vm_name=vm_name, resource_group_name=resource_group_name, exc=exc))
+
+
+def stop_virtual_machine(resource_group_name, vm_name, dry_run=False):
+    if dry_run:
+        LOGGER.info("[DRY RUN] Would stop VM: {vm_name} in resource group: {resource_group_name}".format(
+            vm_name=vm_name, resource_group_name=resource_group_name))
+    else:
+        LOGGER.info("Stopping VM: {vm_name} in resource group: {resource_group_name}".format(
+            vm_name=vm_name, resource_group_name=resource_group_name))
+        try:
+            compute_client.virtual_machines.begin_deallocate(resource_group_name, vm_name)
+        except Exception as exc:  # pylint: disable=broad-except
+            LOGGER.info("Failed to stop VM: {vm_name} in resource group: {resource_group_name} with exception: {exc}".format(
+                vm_name=vm_name, resource_group_name=resource_group_name, exc=exc))
+
+
+def delete_resource_group(resource_group_name, dry_run=False):
+    if dry_run:
+        LOGGER.info("[DRY RUN] Would delete resource group: {resource_group_name}".format(
+            resource_group_name=resource_group_name))
+    else:
+        LOGGER.info("Deleting resource group: {resource_group_name}".format(resource_group_name=resource_group_name))
+        try:
+            resource_client.resource_groups.begin_delete(resource_group_name)
+        except Exception as exc:  # pylint: disable=broad-except
+            LOGGER.info("Failed to delete resource group: {resource_group_name} with exception: {exc}".format(
```

Comment:
I am not sure how linter doesn't complain about such lines.
But in common case a `LOGGER` line must use lazy formatting:
```
LOGGER.info("foo: %s, then %s", a, b)
```